clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -872,6 +875,11 @@ public void subscribe(Pattern pattern, ConsumerRebalanceListener listener) {
     public void unsubscribe() {
         acquire();
         try {
+            if (!this.subscriptions.allConsumed().isEmpty() && this.autoCommitEnabled)
```

Comment:
I am running into an issue with your suggestion in the second bullet. The method `maybeAutoCommitOffsetsAsync(...)` takes a `long`, and when I pass in `time.milliseconds()` (i.e. calling `this.coordinator.maybeAutoCommitOffsetsAsync(time.milliseconds());` here) it doesn't execute the main block of code because the condition `now >= nextAutoCommitDeadline` does not hold. It seems we need to pass in a `long` value that's greater than the next auto-commit time (in milliseconds). Since we just want to make sure the main block of code in that method (which involves committing the offsets) is executed, we could just pass `Long.MAX_VALUE`; though I'm not sure if that's the cleanest approach. What's your opinion?